lib/IRGen/Fulfillment.cpp

```diff
@@ -284,16 +284,24 @@ bool FulfillmentMap::searchNominalTypeMetadata(IRGenModule &IGM,
     }
 
     // Otherwise, it's a conformance.
+    ProtocolConformanceRef conf = *confOrNone;
 
     // Ignore it unless the type itself is interesting.
     if (!keys.isInterestingType(arg))
       return;
 
+    if (conf.isInvalid()) {
```

Comment:
Generally, we don't expect to see invalid conformances in valid code. The fact that this is coming up here suggests that the bug might be in a lower layer, perhaps in the implementation of `enumerateFulfilments()`, or in the `GenericSignatureBuilder` itself.

Reply:
Right.  Which means this might be a very subtle and difficult bug to fix — if you're inclined to try  anyway, that's great, but don't feel bad about deciding that you should leave it alone for now.